cmd/kubeadm/app/cmd/config.go

```diff
@@ -335,15 +339,17 @@ func NewCmdConfigUploadFromFile(out io.Writer, kubeConfigFile *string) *cobra.Co
 }
 
 // NewCmdConfigUploadFromFlags returns cobra.Command for "kubeadm config upload from-flags" command
+// Deprecated: please see kubeadm init phase upload-config
 func NewCmdConfigUploadFromFlags(out io.Writer, kubeConfigFile *string) *cobra.Command {
 	cfg := &kubeadmapiv1beta2.InitConfiguration{}
 	kubeadmscheme.Scheme.Default(cfg)
 
 	var featureGatesString string
 
 	cmd := &cobra.Command{
-		Use:   "from-flags",
-		Short: "Create the in-cluster configuration file for the first time from using flags",
+		Deprecated: "please see kubeadm init phase upload-config",
+		Use:        "from-flags",
+		Short:      "Create the in-cluster configuration file for the first time from using flags",
```

Comment:
just wanted to note that potential consumers of this variant would need to migrate to file only.
also that we need to have this deprecated for 12 months as GA functionality.
cc @rosti

Reply:
@neolit123 do you want me to put that in the above deprecation message?

Reply:
let's wait on rosti's comments.

Reply:
This is actually the worst of them all. You don't get to see what's getting uploaded and users can easily mistake this to "patch" sort of command. Thus, they can inadvertently set some fields of their ClusterConfiguration to defaults and not even notice it.